### ISSUE TITLE
[Issue]: Fail to import RetrieveUserProxyAgent - ImportError: Please install dependencies first. 

### ISSUE THREAD

**Issue by Nilbug** [microsoft/autogen#2293]
### Describe the issue

Failed to import RetrieveUserProxyAgent with ImportError: Please install dependencies first. pip install pyautogen[retrievechat].  But I can import RetrieveAssistantAgent without error. 

I found a similar issue on github https://github.com/microsoft/autogen/issues/1220 and I tried to install different versions of pyautogen (0.2.6, 0.2.20 and 0.2.21) but none works. 

Also, there seems to be another error(also in the screenshot):
from google.protobuf import descriptor as _descriptor
ModuleNotFoundError: No module named 'google.protobuf'; 'google' is not a package

### Steps to reproduce

step 1: pip install "pyautogen[retrievechat]" (the installed version is 0.2.21)
step 2:  run the code:
import autogen
from autogen.agentchat.contrib.retrieve_user_proxy_agent import RetrieveUserProxyAgent

### Screenshots and logs

![Screenshot 2024-04-05 at 10 52 16 PM](https://github.com/microsoft/autogen/assets/70528707/0acd7f61-5b18-41cc-8dd3-af270e49f1d4)

### Additional Information

AutoGen Version: 0.2.21
Operating System: MacOS
Python Version: 3.11.3

**Comment by thinkall** [comment]
Thanks for the feedback @Nilbug . Did you try with a new virtual env or conda env? What's your chromadb version, does simply  `import chromadb` works in your env?

**Comment by Nilbug** [comment]
Thanks a lot for your reply!
I tried a new conda env, with python 3.11.8 but the same issue persists. 

Import chromadb gives ModuleNotFoundError: No module named 'google.protobuf'; 'google' is not a package. I tried to pip uninstall and reinstall protobuf, but nothing changed. there is clearly 'google.protobuf' in the env.
![Screenshot 2024-04-06 at 2 40 53 PM](https://github.com/microsoft/autogen/assets/70528707/ac8eb86e-ec12-4d11-a4e1-3e5a51c35848)

Then I tried "conda install -c anaconda protobuf" and now everything worked, I can import RetrieveUserProxyAgent successfully. Thanks a lot!
